pkg/kubectl/cmd/create.go

```diff
@@ -172,3 +181,54 @@ func createAndRefresh(info *resource.Info) error {
 	info.Refresh(obj, true)
 	return nil
 }
+
+// nameFromCommandArgs is a utility function for commands that assume the first argument is a resource name
+func nameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
+	if len(args) == 0 {
+		return "", cmdutil.UsageError(cmd, "NAME is required")
+	}
+	return args[0], nil
+}
+
+// createSubcommandInternal is intended to be used by subcommands that generator client side objects
+func createSubcommandInternal(f *cmdutil.Factory, cmd *cobra.Command, name string, out io.Writer, structuredGenerator kubectl.StructuredGenerator) error {
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	obj, err := structuredGenerator.StructuredGenerate()
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
+	gvk, err := typer.ObjectKind(obj)
+	mapping, err := mapper.RESTMapping(unversioned.GroupKind{Group: gvk.Group, Kind: gvk.Kind}, gvk.Version)
+	if err != nil {
+		return err
+	}
+	client, err := f.RESTClient(mapping)
+	if err != nil {
+		return err
+	}
+	if !cmdutil.GetFlagBool(cmd, "dry-run") {
+		resourceMapper := &resource.Mapper{ObjectTyper: typer, RESTMapper: mapper, ClientMapper: f.ClientMapperForCommand()}
+		info, err := resourceMapper.InfoForObject(obj)
+		if err != nil {
+			return err
+		}
+		// Serialize the configuration into an annotation.
+		if err := kubectl.UpdateApplyAnnotation(info); err != nil {
+			return err
+		}
+		obj, err = resource.NewHelper(client, mapping).Create(namespace, false, info.Object)
+		if err != nil {
+			return err
+		}
+	}
+	outputFormat := cmdutil.GetFlagString(cmd, "output")
+	if outputFormat != "" {
+		return f.PrintObject(cmd, obj, out)
```

Comment:
This is why you need command, but I don't think that's enough reason to make a non-composeable function for your other flag values.  `PrintObject` is something we also need to refactor to be re-useable downstream (its caused us multiple runtime panics), but I don't think it has to be done for this pull.